monai/transforms/spatial/array.py

```diff
@@ -990,7 +990,8 @@ def __call__(
                 affine = affine @ create_scale(spatial_dims, self.scale_params)
             self.affine = affine
 
-        self.affine = torch.as_tensor(np.ascontiguousarray(self.affine), device=self.device)
+        if not isinstance(self.affine, torch.Tensor):
+            self.affine = torch.as_tensor(np.ascontiguousarray(self.affine), device=self.device)
```

Comment:
I think this one will address #1827  what do you think? @Nic-Ma

Reply:
Hi @wyli ,

I already submitted another PR to fix this issue https://github.com/Project-MONAI/MONAI/pull/1836, we should not use `self.affine` here as it's not thread-safe and may cause issues in `CacheDataset` caching.

Thanks.

Reply:
i think the update of the variable is not shared among process, should be ok, or do you have a test case to show the issue?

Reply:
It should be OK in DataLoader because DataLoader is multi-processing, but it may cause error if you want to cache the Affined transform results in `CacheDataset` with `num_workers > 0` as it's multi-threads logic, right?

Reply:
ok, I think we could
- non-random transforms shouldn't write their instance members
- random transforms could write instance variables
- try to make cache dataset calls threadsafe

Reply:
Let me try to update my PR.